examples/usercall-extension-bind/runner/src/main.rs

```diff
@@ -0,0 +1,356 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate aesm_client;
+extern crate enclave_runner;
+extern crate sgxs_loaders;
+extern crate byteorder;
+extern crate libc;
+
+use aesm_client::AesmClient;
+use enclave_runner::usercalls::{SyncListener, SyncStream, UsercallExtension};
+use enclave_runner::EnclaveBuilder;
+use sgxs_loaders::isgx::Device as IsgxDevice;
+
+use std::io::{Result as IoResult, Error, ErrorKind, Read, Write};
+use std::net::Shutdown;
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr, TcpListener, TcpStream};
+use byteorder::{NetworkEndian, ReadBytesExt};
+use std::thread;
+use std::mem::size_of;
+
+/// This example demonstrates use of usercall extensions for bind call.
+/// User call extension allow the enclave code to "bind" to an external service via a customized enclave runner.
+/// Here we customize the runner to intercept calls to bind to an address and advance the stream before returning it to enclave
+/// This can be useful to strip protocol encapsulations, say while servicing requests load balanced by HA Proxy.
+/// This example demonstrates de-encapsulation for various HA proxy configurations before handing over the stream to the enclave.
+/// To simulate HA proxy configurations, the runner spawns a thread that connects to the same address which enclave binds to and 
+/// writes encapsulated test data for various HA proxy configurations to the stream.
+
+
+const SIZE_HEADER_SIG: usize = 12;
+static HEADER_SIG: [u8; SIZE_HEADER_SIG] = [0x0d, 0x0a, 0x0d, 0x0a, 0x00, 0x0d, 0x0a, 0x51, 0x55, 0x49, 0x54, 0x0a];
+const V2CMD_LOCAL: u8 = 0x2_0;
+const V2CMD_PROXY: u8 = 0x2_1;
+const FAMILY_UNSPEC: u8 = 0;
+const FAMILY_TCP4: u8 = 0x11;
+const FAMILY_TCP6: u8 = 0x21;
+
+
+struct ProxyIpV4Addr {
```

Comment:
Rust convention is to use a lowercase `v` here https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html (same for v6)

Reply:
Incorporated in recent push.